--- target/linux/ramips/dts/HC5962.dts	2020-05-05 11:01:46.825136800 +0800
+++ target/linux/ramips/dts/HC5962.dts	2020-05-03 21:02:47.000000000 +0800
@@ -16,11 +16,6 @@
 		led-upgrade = &led_status;
 	};
 
-	memory@0 {
-		device_type = "memory";
-		reg = <0x0 0x10000000>;
-	};
-
 	chosen {
 		bootargs = "console=ttyS0,115200";
 	};
@@ -41,8 +36,7 @@
 	};
 
 	keys {
-		compatible = "gpio-keys-polled";
-		poll-interval = <20>;
+		compatible = "gpio-keys";
 
 		reset {
 			label = "reset";
@@ -50,6 +44,22 @@
 			linux,code = <KEY_RESTART>;
 		};
 	};
+
+	ubi-concat {
+		compatible = "mtd-concat";
+		devices = <&ubipart0 &ubipart1>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			ubi@540000 {
+				label = "ubi";
+				reg = <0x0 0x79c0000>;
+			};
+		};
+	};
 };
 
 &nand {
@@ -80,18 +90,12 @@
 
 		partition@140000 {
 			label = "kernel";
-			reg = <0x140000 0x200000>;
+			reg = <0x140000 0x400000>;
 		};
 
-		partition@340000 {
-			label = "ubi";
-			reg = <0x340000 0x1E00000>;
-		};
-
-		partition@2140000 {
-			label = "hw_panic";
-			reg = <0x2140000 0x80000>;
-			read-only;
+		ubipart0: partition@540000 {
+			label = "ubipart0";
+			reg = <0x540000 0x1c80000>;
 		};
 
 		partition@21c0000 {
@@ -100,30 +104,9 @@
 			read-only;
 		};
 
-		partition@2240000 {
-			label = "backup";
-			reg = <0x2240000 0x80000>;
-			read-only;
-		};
-
-		partition@22c0000 {
-			label = "overly";
-			reg = <0x22c0000 0x1000000>;
-		};
-
-		partition@32c0000 {
-			label = "firmware_backup";
-			reg = <0x32c0000 0x2000000>;
-		};
-
-		partition@52c0000 {
-			label = "oem";
-			reg = <0x52c0000 0x200000>;
-		};
-
-		partition@54c0000 {
-			label = "opt";
-			reg = <0x54c0000 0x2ac0000>;
+		ubipart1: partition@2240000 {
+			label = "ubipart1";
+			reg = <0x2240000 0x5d40000>;
 		};
 	};
 };
@@ -148,11 +131,9 @@
 	};
 };
 
-&pinctrl {
-	state_default: pinctrl0 {
-		gpio {
-			ralink,group = "uart3", "jtag";
-			ralink,function = "gpio";
-		};
+&state_default {
+	gpio {
+		ralink,group = "uart3", "jtag";
+		ralink,function = "gpio";
 	};
 };
