--- target/linux/ramips/image/mt7621.mk	2020-05-05 11:03:31.855101100 +0800
+++ target/linux/ramips/image/mt7621.mk	2020-05-04 13:03:08.000000000 +0800
@@ -224,9 +224,9 @@
   DTS := HC5962
   BLOCKSIZE := 128k
   PAGESIZE := 2048
-  KERNEL_SIZE := 2097152
+  KERNEL_SIZE := 4096k
   UBINIZE_OPTS := -E 5
-  IMAGE_SIZE := $(ralink_default_fw_size_32M)
+  IMAGE_SIZE := 32768k
   IMAGES += factory.bin
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
   IMAGE/factory.bin := append-kernel | pad-to $$(KERNEL_SIZE) | append-ubi | check-size $$$$(IMAGE_SIZE)
