--- feeds.conf.default	2020-05-01 16:32:09.821003700 +0800
+++ feeds.conf.default	2020-05-01 16:32:59.071785800 +0800
@@ -1,5 +1,5 @@
 src-git packages https://github.com/coolsnowwolf/packages
 src-git luci https://github.com/coolsnowwolf/luci
 src-git routing https://git.openwrt.org/feed/routing.git;openwrt-19.07
-#src-git telephony https://git.openwrt.org/feed/telephony.git;openwrt-19.07
-#src-git helloworld https://github.com/fw876/helloworld
\ No newline at end of file
+src-git helloworld https://github.com/fw876/helloworld
+src-git lienol https://github.com/Lienol/openwrt-package
