--- package/network/services/hostapd/files/hostapd.sh	2020-05-01 16:41:55.190378300 +0800
+++ package/network/services/hostapd/files/hostapd.sh	2020-05-01 16:42:03.711482500 +0800
@@ -1077,8 +1077,8 @@
 
 	_wpa_supplicant_common "$ifname"
 
-	ubus wait_for wpa_supplicant.$phy
-	ubus call wpa_supplicant.$phy config_add "{ \
+	ubus wait_for wpa_supplicant
+	ubus call wpa_supplicant config_add "{ \
 		\"driver\": \"${_w_driver:-wext}\", \"ctrl\": \"$_rpath\", \
 		\"iface\": \"$ifname\", \"config\": \"$_config\" \
 		${network_bridge:+, \"bridge\": \"$network_bridge\"} \
@@ -1089,7 +1089,7 @@
 
 	[ "$ret" != 0 ] && wireless_setup_vif_failed WPA_SUPPLICANT_FAILED
 
-	local supplicant_pid=$(ubus call service list '{"name": "hostapd"}' | jsonfilter -l 1 -e "@['hostapd'].instances['supplicant-${phy}'].pid")
+	local supplicant_pid=$(ubus call service list '{"name": "hostapd"}' | jsonfilter -l 1 -e "@['hostapd'].instances['supplicant'].pid")
 	wireless_add_process "$supplicant_pid" "/usr/sbin/wpa_supplicant" 1
 
 	return $ret